_posts/2023-07-17-elytron-oidc-client-scope.adoc

```diff
@@ -0,0 +1,153 @@
+---
+layout: post
+title: 'Configuring Elytron OIDC Client with Additional Scope Values'
+date: 2023-07-17
+tags: oidc authentication 
+synopsis: How to secure a web application deployed to WildFly with OpenID Connect using additional scope values. 
+author: prarthonapaul
+---
+
+:toc: macro
+:toc-title:
+
+https://openid.net/developers/how-connect-works/[ OpenID Connect] is an identity layer on top of the OAuth 2.0 protocol that makes it possible for a client to verify a user’s identity based on authentication that’s performed by an OpenID provider. The https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest[OpenID Connect specifications] indicate that there are other scope values which may be included in the authentication request to ask permission to additional and specific resources.The purpose of this example is to demonstrate how to configure additional scope values to request access to additional fields.
+
+toc::[]
+
+Note: This quickstart demonstrates how to configure scope values in two different ways: 
+
+* Deployment Configuration
+* Subsystem Configuration
+
+== Setting up Keycloak OpenID Provider
+
+It’s easy to set up Keycloak using Docker. Follow the steps in Keycloak’s https://www.keycloak.org/getting-started/getting-started-docker[ getting started guide ] to start Keycloak, create a realm called `myrealm` and register a client called `myclient`.
+
+After registering our client, `myclient`, we also need to configure valid redirect URIs. Simply click on `Clients` and then on `myclient`. In the `Valid Redirect URIs field`, enter http://localhost:8090/simple-webapp-oidc/*.
+
+Navigate to the `client scopes` tab for `myclient` and change the `assigned-type` for all scope values from `default` to `optional`. This ensures that you will only get access to claims that you request using the scope values you configure. 
```

Comment:
I think we should try to match the names from the Keycloak UI, including case, e.g.,

`Client scopes` instead of client scopes

`Assigned type` instead of assigned-type

Reply:
"This ensures that you will only get access to claims that you request using the scope values you configure. "

Might be good to reword/explain this a bit more:

e.g., what happens when assigned type is set to default? how does that differ from when it's set to optional?